library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> Why not? 

It is not giving any extra information to the elaborator. The elaborator already knows the expected type.
The error message will be very similar: expression has type `T'`, but is expected to have `T`.
From the user point of view, this error is weird because she provided a `T`. The bug is in the parser action which produced something of type `T'`. This kind of programming mistake always confuses users. As I said above, I'm fine with this limitation of the weakly typed approach. If this happens, users will complain (as they complain about bugs in Lean), and the buggy parser action will be fixed.